File: Backend/app/dtos/users/admin_user_dto.py

from app.utils.objectid import ObjectId
from pydantic import Field, RootModel
from app.dtos.users.user_response_dto import UserResponseDataDTO
from app.utils.pyobjectid import PyObjectId



class AdminCreateUserDataDTO(UserResponseDataDTO):
    created_by: PyObjectId = Field(... ,alias="created_by_admin_id")

class AdminUpdateUserDataDTO(UserResponseDataDTO):
    updated_by: PyObjectId = Field(..., alias="updated_by_admin_id")

class AdminDeleteUserDataDTO(UserResponseDataDTO):
    deleted_by: PyObjectId = Field(..., alias="deleted_by_admin_id")

class AdminFindUserDataDTO(UserResponseDataDTO):
    created_by: PyObjectId = Field(..., alias="created_by_admin_id")
    updated_by: PyObjectId = Field(..., alias="updated_by_admin_id")
    deleted_by: PyObjectId = Field(..., alias="deleted_by_admin_id")

    model_config = {
        "from_attributes": True,
        "json_encoders": {ObjectId: str, PyObjectId: str},
        "arbitrary_types_allowed": True,
        "extra": "ignore",
        "populate_by_name": True, 
    }

class AdminFindUserDataListDTO(RootModel[list[AdminFindUserDataDTO]]):
    model_config = {"from_attributes": True}

File: Frontend/src/types/dtos/users/admin_user_dto.ts
import type { UserResponseDataDTO } from "./user_action_dto";

export interface AdminCreateUserDataDTO extends UserResponseDataDTO {
  created_by: string;
}

export interface AdminUpdateUserDataDTO extends UserResponseDataDTO {
  updated_by: string;
}

export interface AdminDeleteUserDataDTO extends UserResponseDataDTO {
  deleted_by: string;
}

export interface AdminFindUserDataDTO extends UserResponseDataDTO {
  created_by: string;
  updated_by: string;
  deleted_by: string;
}
